### PR TITLE
Added zookeeper 3.4.7 support

### PR DESCRIPTION
**Pull request by bpanneton**

### CODE REVIEW

**Comment by bpanneton** [comment]
I wasn't able to test this with the testsuite but my normal runs seemed to work fine. Once I get lsf support working for the testsuite I will verify it is working properly.

File: testsuite/test-generate.sh [hunk 1]
```diff
@@ -1086,7 +1087,7 @@ for hbaseversion in 0.98.3-hadoop2 0.98.9-hadoop2
 do
     for hadoopversion in 2.6.0
     do
-	for zookeeperversion in 3.4.6
+	for zookeeperversion in 3.4.6 3.4.7
```
Comment (chu11) [inline testsuite/test-generate.sh]: I'd like to limit the permutation of zookeeper tests in non-zookeeper sections.  How about we leave this one as 3.4.6 (b/c its running against an older hadoop) ... and ... (see below)

File: testsuite/test-generate.sh [hunk 2]
```diff
@@ -1153,7 +1154,7 @@ for hbaseversion in 0.99.0 0.99.1 0.99.2 1.0.0 1.0.1 1.0.1.1 1.0.2 1.1.0 1.1.0.1
 do
     for hadoopversion in 2.7.0
     do
-	for zookeeperversion in 3.4.6
+	for zookeeperversion in 3.4.6 3.4.7
```
Comment (chu11) [inline testsuite/test-generate.sh]: make this 3.4.7 only.

File: testsuite/test-generate.sh [hunk 3]
```diff
@@ -1222,7 +1223,7 @@ for hbaseversion in 0.98.3-hadoop2 0.98.9-hadoop2
 do
     for hadoopversion in 2.6.0
     do
-	for zookeeperversion in 3.4.6
+	for zookeeperversion in 3.4.6 3.4.7
```
Comment (chu11) [inline testsuite/test-generate.sh]: similar to what I said about

File: testsuite/test-generate.sh [hunk 4]
```diff
@@ -1244,7 +1245,7 @@ for hbaseversion in 0.99.0 0.99.1 0.99.2 1.0.0 1.0.1 1.0.1.1 1.0.2 1.1.0 1.1.0.1
 do
     for hadoopversion in 2.7.0
     do
-	for zookeeperversion in 3.4.6
+	for zookeeperversion in 3.4.6 3.4.7
 	do
```
Comment (chu11) [inline testsuite/test-generate.sh]: similar to what I said about

File: testsuite/test-generate.sh [hunk 5]
```diff
@@ -1268,7 +1269,7 @@ for hbaseversion in 0.98.3-hadoop2 0.98.9-hadoop2
 do
     for hadoopversion in 2.6.0
     do
-	for zookeeperversion in 3.4.6
+	for zookeeperversion in 3.4.6 3.4.7
```
Comment (chu11) [inline testsuite/test-generate.sh]: similar to what I said about

**Comment by chu11** [comment]
Also, could you run a 'make' on script-templates to generate the new templates?  That could be a new commit, so that the first patch set is relatively clean.

**Comment by bpanneton** [comment]
Wasn't sure what to do with storm, so I left them both. Let me know if you want me to remove them.

File: testsuite/test-generate.sh [hunk 6]
```diff
@@ -1086,7 +1087,7 @@ for hbaseversion in 0.98.3-hadoop2 0.98.9-hadoop2
 do
     for hadoopversion in 2.6.0
     do
-	for zookeeperversion in 3.4.6
+	for zookeeperversion in 3.4.6 
 	do
```
Comment (chu11) [inline testsuite/test-generate.sh]: could you clean up the whitespace? :-)  It's in a few places it seems.

File: testsuite/test-generate.sh [hunk 7]
```diff
@@ -1870,7 +1871,7 @@ echo "Making Storm tests"
 
 for stormversion in 0.9.3 0.9.4
 do
-    for zookeeperversion in 3.4.6
+    for zookeeperversion in 3.4.6 3.4.7
```
Comment (chu11) [inline testsuite/test-generate.sh]: lets just to 3.4.6 for this one.

File: testsuite/test-generate.sh [hunk 8]
```diff
@@ -1902,7 +1903,7 @@ done
 
 for stormversion in 0.9.5
 do
-    for zookeeperversion in 3.4.6
+    for zookeeperversion in 3.4.6 3.4.7
```
Comment (chu11) [inline testsuite/test-generate.sh]: and just 3.4.7 here.

File: testsuite/test-submit.sh [hunk 9]
```diff
@@ -474,7 +474,7 @@ then
     do
         for hadoopversion in 2.6.0
         do
-            for zookeeperversion in 3.4.6
+            for zookeeperversion in 3.4.6 3.4.7
```
Comment (chu11) [inline testsuite/test-submit.sh]: i think a bunch of these need to be adjusted given above changes?

**Comment by bpanneton** [comment]
Think I got them all.

**Comment by chu11** [comment]
Did you not push the right patch, some of the diffs above still seem to exist.

**Comment by bpanneton** [comment]
ah! I didn't commit them all before I pushed. Check now!

**Comment by chu11** [comment]
LGTM, and good catch on the accidental duplicate code.

**Pull request merged**